src/Storages/StorageMySQL.cpp

```diff
@@ -281,7 +281,7 @@ StorageMySQLConfiguration StorageMySQL::getConfiguration(ASTs engine_args, Conte
         configuration.table = engine_args[2]->as<ASTLiteral &>().value.safeGet<String>();
         configuration.username = engine_args[3]->as<ASTLiteral &>().value.safeGet<String>();
         configuration.password = engine_args[4]->as<ASTLiteral &>().value.safeGet<String>();
-
+        context_->getRemoteHostFilter().checkHostAndPort(configuration.addresses[0].first, toString(configuration.addresses[0].second));
```

Comment:
Better to move this check after line 289, because otherwise it is skipped for `named_collections` case.
Also worth to iterate addresses and check each, not only the first.

Reply:
@kssenii , named_collection should be safe to access ,Since it is in config.xml .
Do we really need to check for addresses defined under <named_collections>?